### DISCUSSION TITLE
Improvement of payment memo display on Mixin Messenger

### DISCUSSION THREAD

**Discussion by zed-wong** [MixinNetwork/mixin#153] [Ideas]
When [invoking a payment](https://developers.mixin.one/docs/schema#payment), the memo would be shown on the payment page of Mixin Messenger. In most cases, the memo is encoded and unreadable by non-technical users, so it's kind of worthless to them. And since there's always a clear sign to let users know what they're doing. It's the only part that might possibly confuse normal users.

- Exinone:
<img src="https://user-images.githubusercontent.com/36271125/218238339-6a0f5ae8-f3bf-4ee7-9fb4-5519bfa3592e.jpg" width="265" height="425">

- 4swap:
<img src="https://user-images.githubusercontent.com/36271125/218238294-9f5e99b8-ce35-4171-b5c4-5cd80737f796.jpg"
width="242" height="425">

My suggestion is to add a parameter for [invoking payment](https://developers.mixin.one/docs/schema#payment) to show a message from the app that tells people what they are doing. And that message shows up when invoking a payment. Also, the memo could be shown by clicking on this message or an icon.

This proposal is inspired by Rabby wallet and Phantom wallet in their comparison to Metamask.
https://twitter.com/phantom/status/1624139794661134337
https://twitter.com/Rabby_io/status/1620003926555922432

**Comment by cedricfung** [comment]
A readable memo is needed, we could create some standard memo protocol, but encryption should also be considered together, e.g. all Pando apps encrypt the memo for privacy reason.
